domain/user/state/state.go

```diff
@@ -429,6 +429,51 @@ func (st *State) SetActivationKey(ctx context.Context, name string, activationKe
 	})
 }
 
+// GetActivationKey retrieves the activation key for the user with the supplied
+// user name. If the user does not exist an error that satisfies
+// usererrors.NotFound will be returned.
+func (st *State) GetActivationKey(ctx context.Context, name string) ([]byte, error) {
+	db, err := st.DB()
+	if err != nil {
+		return nil, errors.Annotate(err, "getting DB access")
+	}
+
+	uuidStmt, err := st.getActiveUUIDStmt()
+	if err != nil {
+		return nil, errors.Trace(err)
+	}
+
+	m := make(sqlair.M, 1)
+
+	selectKeyStmt, err := sqlair.Prepare("SELECT (*) AS (&ActivationKey.*) FROM user_activation_key WHERE user_uuid = $M.uuid", m, ActivationKey{})
+	if err != nil {
+		return nil, errors.Annotate(err, "preparing activation get query")
+	}
+
+	var key ActivationKey
```

Comment:
In any case, if you declare this above prepare you could use it for the type info.